controllers/configurationpolicy_controller.go

```diff
@@ -1606,21 +1618,42 @@ func (r *ConfigurationPolicyReconciler) handleObjects(
 			} else {
 				resultEvent.compliant = true
 				resultEvent.reason = reasonWantNotFoundDNE
+
+				// Compliant, complianceType == mustnothave and the name field is empty
+				// Find resources in the same namespace to add to the status.relatedObjects
+				if objDetails.kind != "" && objDetails.name == "" {
+					if len(allResourceNames) > 0 {
+						shouldAddCondensedRelatedObj = true
```

Comment:
Same comment as above. I don't think this accounts for unnamed object templates when `getNamesOfKind` returns no objects. In this case, just removing the `if len(allResourceNames) > 0 {` conditional should fix that.

Reply:
So if there is no noncompliant resource, we present as * then when the user clicks *, query nothing?

Reply:
Yeah, I think so. It just shows the query that was used for considering objects.